AmplifyPlugins/API/AWSAPICategoryPluginTests/AWSAPICategoryPluginTestBase.swift

```diff
@@ -71,7 +72,12 @@ class AWSAPICategoryPluginTestBase: XCTestCase {
         }
     }
 
-    override func tearDown() {
-        Amplify.reset()
```

Comment:
Will it fix the problem with the data race if we kept the `Amplify.reset()` in `tearDown`, added the `wait(for: 1)` here and remove `Amplify.reset()` from the `setUp()`?

Reply:
i think this means some tests are not calling `Amplify.reset()` on tearDown, which could happen if not all tests are using this test base. In that case, then calling reset in the setUp of this test makes sense since we can't guaranteed  that other tests are resetting properly